File: dotnet/system/workspace/typescript/libs/workspace/domain/system/src/lib/database/iresult.ts
import { IObject } from '../IObject';
import { IDerivationError } from './IDerivationError';

export interface IResult {
  errorMessage: string;

  versionErrors: IObject[];

  accessErrors: IObject[];

  missingErrors: IObject[];

  derivationErrors: IDerivationError[];
}

File: dotnet/system/workspace/csharp/allors.workspace.domain/api/iresult.cs
// <copyright file="IWorkspace.cs" company="Allors bvba">
// Copyright (c) Allors bvba. All rights reserved.
// Licensed under the LGPL license. See LICENSE file in the project root for full license information.
// </copyright>

namespace Allors.Workspace
{
    using System.Collections.Generic;

    public interface IResult
    {
        bool HasErrors { get; }

        string ErrorMessage { get; }

        IEnumerable<IObject> VersionErrors { get; }

        IEnumerable<IObject> AccessErrors { get; }

        IEnumerable<IObject> MissingErrors { get; }

        IEnumerable<IDerivationError> DerivationErrors { get; }
    }
}
